lib/orogen/gen/typekit.rb

```diff
@@ -1499,6 +1499,13 @@ def perform_pending_loads
                 # earlier) so that the importer can fill-in some of the
                 # metadata, like doc and source_file_line.
                 options[:include_paths] = include_dirs
+                
+                #if the user defined additional compiler defines, hand them over to the importer 
+                #because defines might change how the code looks (e.g. some methods might be ifdef guarded)
+                unless preprocess_options[:define].nil? 
```

Comment:
Matters of style in orogen (well, any Ruby code that I maintain):
 - don't use 'unless', replace with if !
 - unless you are testing a tri-state variable (where 'false' and 'nil' are different), don't test against nil explicitely

In other words, here,

```
if !preprocess_options[:define]
end
```

Finally, :define *is set* unconditionally, so why are you checking if it is ?